src/status.js

```diff
@@ -254,62 +263,268 @@
+					if (data == null) {
+						data = []
+					}
+					this._partReferenceMap = data
+					resolve() // returns a map of partid to its properties
+				})
+			})
+		}
+		Logger.debug('PartReferenceMap already exists, no need to request. Number of parts:', this._partReferenceMap.length)
+	};
+
+	/**
+	 * Get 'StatusEvts' reference map to reduce status payload. Duplicated contents in status are stored in the map.
+	 */
+	Status.prototype._getStatusEvtReferenceMap = function () {
+		if (this._statusEvtReferenceMap == null || this._statusEvtReferenceMap.length == 0) {
+			return new Promise((resolve, reject) => {
+				this.selector.request({
+					service: 'Status',
+					func: 'GetStatusEvtReferenceMap',
+					obj: {
+						interface: 'fr.partnering.Status',
+						path: '/fr/partnering/Status'
+					}
+				}, (peerId, err, data) => {
+					Logger.debug(`StatusEvtReferenceMap response`, data, err)
+					if (data == null) {
+						data = []
+					}
+					this._statusEvtReferenceMap = data
+					resolve() // returns a map of partid to its properties
+				})
+			})
+		}
+		Logger.debug('StatusEvtReferenceMap already exists, no need to request. Number of parts:', this._statusEvtReferenceMap.length)
+	};
+
+	/**
+	 * Subscribes to status changes for all parts
+	 * @param {*} parts 
+	 * @param {*} callback 
+	 */
+	Status.prototype._subscribeToStatusChanged = function (parts, callback) {
+		if (parts == null) {
+			return
+		}
+
+		parts.forEach(part => {
+			if (part.objectPath == null) {
+				Logger.warn('Subscribe to StatusChanged: input error', part)
+				return
+			}
+			Logger.debug(`Subscribing to ${part.objectPath}`)
+			let subs = this.selector.subscribe({
+				service: 'status',
+				func: 'StatusChanged',
+				obj: {
+					interface: 'fr.partnering.Status.Part',
+					path: part.objectPath
+				}
+			}, (peerId, err, data) => {
+				if (err != null) {
+					Logger.error("StatusSubscribe:" + err)
+					return
+				}
+				Logger.debug(`Part's StatusChanged is called, data:`, data)
+				if (data[9] == null) data[9] = '' // empty description
+				// Update robotModel variable
+				// Since data is one-dimensional array, make it two-dimensional
+				this._getRobotModelFromRecv2([data], part.RobotId, part.RobotName);
+				if (typeof callback === 'function') {
+					callback(this.robotModel);
+				}
+			})
+			this.subscriptions.push(subs);
+		})
+	}
+
 	/**
+	 * Query for initial statuses
 	 * Subscribe to error/status updates
 	 */
 	Status.prototype.watch = function (robotNames, callback) {
+		Logger.debug(`Status.watch: robotNames`, robotNames)
+
 		this.selector.setMaxListeners(0);
 		this.selector._connection.setMaxListeners(0);
-		let sendData = [];
-		let robotIds = [];
-		return Promise.try(_ => {
-			let req = this.selector.request({
+
+		// Promise to retrieve list of paired neighbors, i.e. all neighbor robots in the same mesh network
+		let getNeighbors = new Promise((resolve, reject) => {
```

Comment:
Not sure this is relevant to filter for robots that are not connected, because we may want to get statuses anyway.
Robots that are not relevant anymore should be blacklisted i.e. already taken into account at diya-node-status side (**normally**)